src/java.desktop/share/classes/javax/print/MimeType.java

```diff
@@ -571,10 +571,9 @@ private void parse(String s) {
             throw new NullPointerException();
         }
         LexicalAnalyzer theLexer = new LexicalAnalyzer (s);
-        int theLexemeType;
-        Vector<String> thePieces = new Vector<>();
-        boolean mediaTypeIsText = false;
-        boolean parameterNameIsCharset = false;
+        ArrayList<String> thePieces = new ArrayList<>();
+        boolean mediaTypeIsText;
+        boolean parameterNameIsCharset;
```

Comment:
Default values might be removed as a separate cleanup.

Reply:
reverted